In [1]:
import pandas as pd

## Data

Predictions!! All made in the same way 

Sotavento may provide observed meteorological data. 

In [6]:
df = pd.read_csv('EjemploDatos.csv')
print('Nº of rows: ', df.shape[0])
df.head()

Nº of rows:  9


prediction date  10u_(44.0, -9.5)  10u_(44.0, -9.375)  \
0  2016-01-01 00:00:00          1.577097            1.675974   
1  2016-01-01 01:00:00          1.320722            1.059857   
2  2016-01-01 02:00:00          1.562336            1.023518   
3  2016-01-01 03:00:00          0.198002            0.025639   
4  2016-01-01 04:00:00          1.604583            1.043792   

   10u_(44.0, -9.25)  10u_(44.0, -9.125)  10u_(44.0, -9.0)  \
0           1.836863            1.821115          1.681467   
1           1.200238            1.276654          1.713910   
2           0.415607            0.335896          0.563801   
3           0.332890           -0.118526         -0.533199   
4           1.073821            0.065032         -0.961823   

   10u_(44.0, -8.875)  10u_(44.0, -8.75)  10u_(44.0, -8.625)  \
0            1.424021           1.190012            1.106394   
1            1.246137           0.725141            0.203412   
2            0.696369           0.487873            0.106770   
3           -1.093990          -1.123653           -1.040646   
4           -1.394685          -1.394074           -1.230012   

   10u_(44.0, -8.5)  ...  vel100_(42.875, -6.0)  vel100_(43.0, -6.0)  \
0          1.108713  ...               4.542353             5.321404   
1          0.215253  ...               4.708214             5.435709   
2         -0.157024  ...               4.553245             5.325067   
3         -0.928341  ...               4.668390             5.177444   
4         -1.500520  ...               4.791137             5.317643   

   vel100_(43.125, -6.0)  vel100_(43.25, -6.0)  vel100_(43.375, -6.0)  \
0               5.971150              5.862873               5.164530   
1               6.092992              5.361253               3.485183   
2               6.018884              4.991258               2.281320   
3               5.716941              4.493564               1.185751   
4               5.648662              4.902998               3.442938   

   vel100_(43.5, -6.0)  vel100_(43.625, -6.0)  vel100_(43.75, -6.0)  \
0             5.391425               6.107658              6.398007   
1             3.988883               4.845162              5.260955   
2             2.807106               3.654994              4.170403   
3             1.837592               3.329041              4.260316   
4             3.724846               4.111763              4.328406   

   vel100_(43.875, -6.0)  vel100_(44.0, -6.0)  
0               6.472929             6.793221  
1               5.464975             5.908251  
2               4.632105             5.078648  
3               4.934864             5.643699  
4               4.885375             5.730442  

[5 rows x 3481 columns]

In [7]:
# number of columns that contain the string 10u
df.columns[df.columns.str.contains('10u')].shape[0]

435

In [8]:
#number of columns that contain the string vel100
df.columns[df.columns.str.contains('vel100')].shape[0]

435

In [21]:
#unique values of the first 4 characters of the column names
df.columns.str[:4].unique()

Index(['pred', '10u_', '10v_', '2t_(', 'sp_(', '100u', '100v', 'vel1'], dtype='object')

In [1]:
#falta diferenciar vel10 y vel 100 
print(8*435+1)

3481


The EjemploDatos.csv contains 3481 columns that can be divided into: 
 - Prediction Date : day and hour 
 
 - 10u             : u component at height 10m
  
 - 10v             : v component at height 10m
 
 - 2t              : temperature at height 2m
 
 - sp              : pressure at surface
 
 - 100u            : u component at height 100m
 
 - 100v            : v component at height 100m
 
 - vel10           : module of the wind at height 10m
 
 - vel100          : module of the wind at height 100m
 

## Target

Observed production at Prediction Date (MWh)

Potencia máxima del parque: 17560Kwh (los datos, ya confirmado, están en Kwh

In [2]:
target_df = pd.read_csv('EjemploTarget.csv',
                        header=None, 
                        names=['Prediction Date','Target (KWh)'])
target_df.head()

Prediction Date  Target (KWh)
0  2016-01-01 00:00:00       4126.70
1  2016-01-01 01:00:00       5688.51
2  2016-01-01 02:00:00       7742.01
3  2016-01-01 03:00:00       7660.50
4  2016-01-01 04:00:00       9730.48